tobac/merge_split.py

```diff
@@ -0,0 +1,477 @@
+        Default is five (5) frames.
+
+    Returns
+    -------
+
+    d : xarray.core.dataset.Dataset
+        xarray dataset of tobac merge/split cells with parent and child designations.
+
+
+    Example usage:
+        d = merge_split(Track)
+        ds = standardize_track_dataset(Track, refl_mask)
+        both_ds = xr.merge([ds, d],compat ='override')
+        both_ds = compress_all(both_ds)
+        both_ds.to_netcdf(os.path.join(savedir,'Track_features_merges.nc'))
+
+    """
+
+    print("this is an update 3")
+    track_groups = TRACK.groupby("cell")
+    cell_ids = {cid: len(v) for cid, v in track_groups.groups.items()}
+    id_data = np.fromiter(cell_ids.keys(), dtype=int)
+    count_data = np.fromiter(cell_ids.values(), dtype=int)
+    all_frames = np.sort(np.unique(TRACK.frame))
+    a_points = list()
+    b_points = list()
+    a_names = list()
+    b_names = list()
+    dist = list()
+
+    if hasattr(TRACK, "longitude"):
+        print("is in lat/lon")
+        for i in id_data:
+            a_pointx = track_groups[i].grid_longitude[-1].values
+            a_pointy = track_groups[i].grid_latitude[-1].values
+            for j in id_data:
+                b_pointx = track_groups[j].grid_longitude[0].values
+                b_pointy = track_groups[j].grid_latitude[0].values
+                d = geodesic((a_pointy, a_pointx), (b_pointy, b_pointx)).m
+                if d <= distance:
+                    a_points.append([a_pointx, a_pointy])
+                    b_points.append([b_pointx, b_pointy])
+                    dist.append(d)
+                    a_names.append(i)
+                    b_names.append(j)
+    else:
+
+        for i in id_data:
+            # print(i)
+            a_pointx = track_groups[i].projection_x_coordinate[-1].values
+            a_pointy = track_groups[i].projection_y_coordinate[-1].values
+            for j in id_data:
+                b_pointx = track_groups[j].projection_x_coordinate[0].values
+                b_pointy = track_groups[j].projection_y_coordinate[0].values
+                d = np.linalg.norm(
+                    np.array((a_pointx, a_pointy)) - np.array((b_pointx, b_pointy))
+                )
+                if d <= distance:
+                    a_points.append([a_pointx, a_pointy])
+                    b_points.append([b_pointx, b_pointy])
+                    dist.append(d)
+                    a_names.append(i)
+                    b_names.append(j)
+
+    id = []
+    for i in range(len(dist) - 1, -1, -1):
+        if a_names[i] == b_names[i]:
+            id.append(i)
+            a_points.pop(i)
+            b_points.pop(i)
+            dist.pop(i)
+            a_names.pop(i)
+            b_names.pop(i)
+        else:
+            continue
+
+    g = Graph()
+    for i in np.arange(len(dist)):
+        g.add_edge(a_names[i], b_names[i], weight=dist[i])
+
+    tree = minimum_spanning_edges(g)
+    tree_list = list(minimum_spanning_edges(g))
```

Comment:
Is there a reason `minimum_spanning_edges` is run twice? can `tree_list` just be `list(tree)`?

Reply:
Reduced it as suggested.